5025 recommend flight over flightsql (#5044)
influxdata/docs-v2: content/influxdb/cloud-dedicated/get-started/query.md

        FROM home
        WHERE time >= '2022-01-02T08:00:00Z'
        AND time <= '2022-01-02T20:00:00Z'`

      iterator, err := client.Query(context.Background(), database, query)

      if err != nil {
        panic(err)
      }

      w := tabwriter.NewWriter(io.Discard, 4, 4, 1, ' ', 0)
      w.Init(os.Stdout, 0, 8, 0, '\t', 0)
      fmt.Fprintln(w, "time\troom\ttemp\thum\tco")

      for iterator.Next() {
        row := iterator.Value()
        day := (row["time"].(arrow.Timestamp)).ToTime(arrow.TimeUnit(arrow.Nanosecond))
        fmt.Fprintf(w, "%s\t%s\t%.2f\t%.2f\t%d\n", day, row["room"], row["temp"], row["hum"], row["co"])
      }

      w.Flush()
      return nil
    }
    ```

    The sample code does the following:
    
    1.  Imports the following packages:

        - `context`
        - `fmt`
        - `io`
        - `os`
        - `text/tabwriter`
        - `github.com/apache/arrow/go/v12/arrow`
        - `github.com/InfluxCommunity/influxdb3-go/influx`

    2.  Defines a `Query()` function that does the following:

        1.  Instantiates `influx.Client` with InfluxDB credentials.
          
            - **HostURL**: your {{% cloud-name %}} cluster URL.

>             - **AuthToken**:  an [API token](/influxdb/cloud-dedicated/admin/tokens/) with _read_  access to the specified database.

Reviewer: ```suggestion
            - **AuthToken**:  a [database token](/influxdb/cloud-dedicated/admin/tokens/) with _read_  access to the specified database.
```

          _For security reasons, we recommend setting this as an environment
          variable rather than including the raw token string._

        2.  Defines a deferred function to close the client after execution.
        3.  Defines a string variable for the SQL query.

        4.  Calls the `influx.Client.query()` method to send the query request with the database name and SQL string. The `query()` method returns an `iterator` for data in the response stream.
        5.  Iterates over rows and prints the data in table format to stdout.

3.  In your editor, open the `main.go` file you created in the
    [Write data section](/influxdb/cloud-serverless/get-started/write/?t=Go#write-line-protocol-to-influxdb) and insert code to call the `Query()` function--for example:

    ```go
    package main

    func main() {	
      WriteLineProtocol()
      Query()
    }
    ```

>     When the `main` package is executed, `main()` will write and query data stored in {{% cloud-name %}}.

Reviewer: ```suggestion
    When the `main` package is executed, `main()` writes and queries data stored in {{% cloud-name %}}.
```
